app/src/main/java/com/kickstarter/ui/adapters/MessagesAdapter.java

```diff
@@ -7,30 +7,68 @@
 import com.kickstarter.R;
 import com.kickstarter.models.Message;
 import com.kickstarter.ui.viewholders.KSViewHolder;
+import com.kickstarter.ui.viewholders.MessageCenterTimestampViewHolder;
 import com.kickstarter.ui.viewholders.MessageViewHolder;
 
+import org.joda.time.DateTime;
+
+import java.util.Collections;
 import java.util.List;
 
-import static java.util.Collections.emptyList;
+import rx.Observable;
 
 public final class MessagesAdapter extends KSAdapter {
+  public MessagesAdapter() {}
 
-  public MessagesAdapter() {
-    addSection(emptyList());
+  private int getLayoutId(final @NonNull SectionRow sectionRow) {
+    if (objectFromSectionRow(sectionRow) instanceof DateTime) {
+      return R.layout.message_center_timestamp_layout;
+    } else if (objectFromSectionRow(sectionRow) instanceof Message) {
+      return R.layout.message_view;
+    }
+    return R.layout.empty_view;
   }
 
   public void messages(final @NonNull List<Message> messages) {
-    setSection(0, messages);
+    // Group messages by start of day.
```

Comment:
coolest thing about this chunk of code? i just refered to [our swift grouping](https://github.com/kickstarter/ios-oss/blob/master/Kickstarter-iOS/DataSources/ProjectActivitiesDataSource.swift#L23-L44) to do the same thing!!